internal/pkg/agent/cmd/diagnostics.go

```diff
@@ -600,3 +628,33 @@ func zipProfs(zw *zip.Writer, pprof map[string][]client.ProcPProf) error {
 	}
 	return nil
 }
+
+func zipMetrics(zw *zip.Writer, metrics *proto.ProcMetricsResponse) error {
+	zf, err := zw.Create("metrics/")
+	if err != nil {
```

Comment:
This is interesting, I believe this is a real bug, I don't think you need to create `zw.Create()` with only the `metrics/`

Reply:
The documentation for zip states that adding a trailing slash creates a directory: https://pkg.go.dev/archive/zip#Writer.Create